sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2863,6 +2871,130 @@ pi_result cuda_piEnqueueMemUnmap(pi_queue command_queue, pi_mem memobj,
   return ret_err;
 }
 
+// USM
+pi_result cuda_piextUSMHostAlloc(void **result_ptr, pi_context context,
+                                 pi_usm_mem_properties *properties, size_t size,
+                                 pi_uint32 alignment) {
+  PI_CHECK_ERROR(cuMemAllocHost(result_ptr, size));
+  assert(*result_ptr % alignment == 0);
+  return PI_SUCCESS;
+}
+
+pi_result cuda_piextUSMDeviceAlloc(void **result_ptr, pi_context context,
+                                   pi_device device,
+                                   pi_usm_mem_properties *properties,
+                                   size_t size, pi_uint32 alignment) {
+  assert(device != nullptr);
+  ScopedContext active(context);
+  PI_CHECK_ERROR(cuMemAlloc((CUdeviceptr *)result_ptr, size));
+  assert(*result_ptr % alignment == 0);
+  return PI_SUCCESS;
+}
+
+pi_result cuda_piextUSMSharedAlloc(void **result_ptr, pi_context context,
+                                   pi_device device,
+                                   pi_usm_mem_properties *properties,
+                                   size_t size, pi_uint32 alignment) {
+  cl::sycl::detail::pi::die("cuda_piextUSMSharedAlloc not implemented");
+  return PI_ERROR_UNKNOWN;
+}
+
+pi_result cuda_piextUSMFree(pi_context context, void *ptr) {
+  pi_result retErr = PI_SUCCESS;
+  unsigned int type;
+  retErr = PI_CHECK_ERROR(cuPointerGetAttribute(
+      &type, CU_POINTER_ATTRIBUTE_MEMORY_TYPE, (CUdeviceptr)ptr));
+  assert(type == CU_MEMORYTYPE_DEVICE || type == CU_MEMORYTYPE_HOST);
+  if (type == CU_MEMORYTYPE_DEVICE) {
+    retErr = PI_CHECK_ERROR(cuMemFree((CUdeviceptr)ptr));
+  }
+  if (type == CU_MEMORYTYPE_HOST) {
+    retErr = PI_CHECK_ERROR(cuMemFreeHost(ptr));
```

Comment:
[`cuMemAllocHost`](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1gdd8311286d2c2691605362c689bc64e0)  doesn't require `cuMemFreeHost`, only [`cuMemHostAlloc`](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g572ca4011bfcb25034888a14d4e035b9) does.

Reply:
The documentation of [cuMemFreeHost](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g62e0fdbe181dab6b1c90fa1a51c7b92c) says

> Frees the memory space pointed to by `p`, which must have been returned by a previous call to `cuMemAllocHost()`.